### PR TITLE
feat: add ping sound (AR-3121)

### PR DESCRIPTION
**Pull request by alexandreferris**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-3121" title="AR-3121" target="_blank">StoryAR-3121</a>  Ping sound in conversation screen
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [X] contains a reference JIRA issue number like `SQPIT-764`
    - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

When in conversation screen and user received ping, no sounds was played.

### Causes (Optional)

Not implemented.

### Solutions

Add the verification for if a message is Ping(`Knock`) and play the ping sound.

#### Test Coverage (Optional)

- [X] I have added automated test to this contribution

#### How to Test

- Open App
- Open Conversation
- MAKE SURE YOU HAVE SOUND ON
- Receive Ping
- Ping sound should be played

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1525/1/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.2(5879791)(5879791)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.2(5879791)(5879791)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/notification/WireNotificationManager.kt [hunk 1]
```diff
@@ -449,6 +458,27 @@ class WireNotificationManager @Inject constructor(
         }
     }
 
+    private fun verifyIfShouldPlayPingSound(
```
Comment (vitorhugods) [inline app/src/main/kotlin/com/wire/android/notification/WireNotificationManager.kt]: `verify` would be a better name if it returned a `boolean` and didn't have any side effects :)

```suggestion
    private fun playPingSoundIfNeeded(
```

File: app/src/main/kotlin/com/wire/android/notification/WireNotificationManager.kt [hunk 2]
```diff
@@ -449,6 +458,27 @@ class WireNotificationManager @Inject constructor(
...
+        currentScreen: CurrentScreen,
+        notifications: List<LocalNotificationConversation>
+    ) {
+        if (currentScreen is CurrentScreen.Conversation) {
+            val conversationId = currentScreen.id
+            val containsPingMessage = notifications
+                .filter { it.id == conversationId }
+                .any {
+                    it.messages.any { message -> message is LocalNotificationMessage.Knock }
+                }
```
Comment (vitorhugods) [inline app/src/main/kotlin/com/wire/android/notification/WireNotificationManager.kt]:
```suggestion
                .any {
                    it.id == conversationId &&
                    it.messages.any { message -> message is LocalNotificationMessage.Knock }
                }
```

TINY performance boost without losing readability imho.
No need to iterate through all the items and then iterate again on the filtered ones.
Just iterate once and that's it.

**Review by vitorhugods — APPROVED** [review]
Looks good to me, just tiny suggestions.

On a side note, I'd love if this kind of logic was detached from notifications and Reloaded could just observe a `pingFlow` from Kalium and make its decision straight in the `ViewModel`.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1525/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.0.2(5912384)(5912384)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.0.2(5912384)(5912384)-staging-compat.apk)

**Pull request merged**